pkg/generic/thrift/json.go

```diff
@@ -115,36 +150,97 @@ func (m *ReadJSON) SetBinaryWithBase64(enable bool) {
 	m.binaryWithBase64 = enable
 }
 
+// SetDynamicGo ...
+func (m *ReadJSON) SetDynamicGo(convOpts, convOptsWithException *conv.Options, msg remote.Message) {
+	m.msg = msg
+	m.useDynamicGo = true
+	if m.isClient {
+		// set dynamicgo option to handle an exception field
+		m.cv = t2j.NewBinaryConv(*convOptsWithException)
+	} else {
+		m.cv = t2j.NewBinaryConv(*convOpts)
+	}
+}
+
 // Read read data from in thrift.TProtocol and convert to json string
 func (m *ReadJSON) Read(ctx context.Context, method string, in thrift.TProtocol) (interface{}, error) {
-	fnDsc, err := m.svc.LookupFunctionByMethod(method)
-	if err != nil {
-		return nil, err
-	}
-	fDsc := fnDsc.Response
-	if !m.isClient {
-		fDsc = fnDsc.Request
-	}
-	resp, err := skipStructReader(ctx, in, fDsc, &readerOption{forJSON: true, throwException: true, binaryWithBase64: m.binaryWithBase64})
-	if err != nil {
-		return nil, err
+	// fallback logic
+	if !m.useDynamicGo {
```

Comment:
long function, consider refactoring the fallback logic to an extra function

Reply:
done